app/src/test/java/com/github/dodobest/upbitapi/UpbitViewModelTest.kt

```diff
@@ -0,0 +1,94 @@
+package com.github.dodobest.upbitapi
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.github.dodobest.data.Injector
+import com.github.dodobest.data.data.UpbitRemoteDataSource
+import com.github.dodobest.domain.InjectorDomain
+import com.github.dodobest.domain.UpbitRepository
+import com.github.dodobest.domain.usecase.GetMarketsUseCase
+import com.github.dodobest.domain.usecase.GetTickerUseCase
+import com.github.dodobest.upbitapi.model.UpbitFakeRemoteDataSet
+import com.github.dodobest.upbitapi.util.getValue
+import com.google.common.truth.Truth.assertThat
+import io.mockk.every
+import io.mockk.mockk
+import io.reactivex.rxjava3.core.Single
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+class UpbitViewModelTest {
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    val testSchedulerRule = RxImmediateSchedulerRule()
+
+    private lateinit var upbitRemoteDataSource: UpbitRemoteDataSource
+    private lateinit var upbitRepository: UpbitRepository
+    private lateinit var getMarketsUseCase: GetMarketsUseCase
+    private lateinit var getTickerUseCase: GetTickerUseCase
+    private lateinit var upbitViewModel: UpbitViewModel
+
+    @Before
+    fun setUp() {
+        upbitRemoteDataSource = mockk()
+        upbitRepository = Injector.provideUpbitRepository(upbitRemoteDataSource)
+        getMarketsUseCase = InjectorDomain.provideGetMarketsUseCase(upbitRepository)
+        getTickerUseCase = InjectorDomain.provideGetTickerUseCase(upbitRepository)
+        upbitViewModel = UpbitViewModel(getMarketsUseCase, getTickerUseCase)
+    }
+
+    @Test
+    fun `UpbitViewModel getMarkets를 호출하면 코인 데이터를 수신한다`() {
+        // given
+        every { upbitRemoteDataSource.getMarkets() } returns Single.just(
+            UpbitFakeRemoteDataSet.upbitMarketData
+        )
+        every { upbitRemoteDataSource.getTicker(BTC_CODE_NAME) } returns Single.just(
+            UpbitFakeRemoteDataSet.upbitBTCTickerData
+        )
+        every { upbitRemoteDataSource.getTicker(ETH_CODE_NAME) } returns Single.just(
+            UpbitFakeRemoteDataSet.upbitETHTickerData
+        )
+        every { upbitRemoteDataSource.getTicker(NU_CODE_NAME) } returns Single.just(
+            UpbitFakeRemoteDataSet.upbitNUTickerData
+        )
+
+        // when
+        upbitViewModel.getMarkets()
+
+        // then
+        assertThat(getValue(upbitViewModel.marketCoinNames)).isEqualTo(
+            UpbitFakeRemoteDataSet.upbitMarketData.map {
+                it.toDomainData()
+            }
+        )
```

Comment:
LiveData의 확장함수로 만들게 되면, `upbitViewModel.marketCoinNames.getOrAwaitValue()` 이렇게 사용할 수 있어요.

```suggestion
        assertThat(getValue(upbitViewModel.marketCoinNames))
            .isEqualTo(
                UpbitFakeRemoteDataSet.upbitMarketData.map { it.toDomainData() }
            )
```
그리고 체이닝 시 가로가 길어지면, 이렇게 쓰시는 것이 더 보기 좋아요. (이하 관련 내용 동일)

ref. https://edu.nextstep.camp/s/OG8R3egF/ls/HpabNSQE

Reply:
확장 함수인 `getOrAwaitValue()`와 그냥 함수 `getValue()` 중 어떤 것을 사용해야하나요?

함수 체이닝 글에서 [아직 Jetpack에서 지원 하지 않아 직접 함수를 작성했다](https://medium.com/androiddevelopers/unit-testing-livedata-and-other-common-observability-problems-bb477262eb04#:~:text=To%20fix%20this%2C%20Jetpack%20doesn%E2%80%99t%20provide%20a%20test%20helper%20yet%2C%20but%20we%20can%20make%20our%20own%3A)고 되어 있더라구요.

[yet](https://issuetracker.google.com/issues/130709056?pli=1)을 눌러보니 함수 체이닝 글을 작성하신 분이 [함수 형식으로 작성한 함수](https://github.com/google/iosched/blob/main/androidTest-shared/src/main/java/com/google/samples/apps/iosched/androidtest/util/LiveDataTestUtil.kt)가 있어서, 이것을 사용했습니다.

Reply:
내부 로직은 동일해서, 가독성 차이일 것 같아요.
그런데 대부분 getOrAwaitValue를 쓰고 있어요.
왜냐하면 getOrAwaitValue는 라이브데이터 값에 접근할 때 .value 대신 사용하는 것이라 어색하지도 않은데, getValue는 파라미터로 라이브데이터를 전달받아야해서 더 보기가 어려워요.

Reply:
[b5027be](https://github.com/DoTheBestMayB/UpbitAPI/pull/21/commits/b5027be524d81308bd28b3a671608fd4475af4a7)
[86ceac9](https://github.com/DoTheBestMayB/UpbitAPI/pull/21/commits/86ceac911775e941fdbcd201f2cbe7e2dbcb7b42)